### PR TITLE
Swift 5 support

### PR DESCRIPTION
**Pull request by djones6**
## Description

Adds a build against a Swift 5 development snapshot to Travis, resolves Swift 5 compilation warnings, and drops support for Swift 3.

#### Compilation warnings

The compiler now emits a warning if a redundant access control modifier is applied to a member of an extension.  This comes about if you declare an explicit access control modifier on the extension itself (ie. `public extension Foo`) and also on members (`public var Bar`).

There are two choices:  keep the default access modifier on the extension (and remove `public` from all members of the extension), or remove the default access modifier and ensure all members that should be `public` are marked as such.

I've chosen the latter, as I think it makes it more clear as to what forms part of our public API: things that are public are still marked as such.

(note: my understanding is that removing the `public` from the extension only affects the _default_ visibility of its members. Public members of the extension remain public.  See: https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html#ID25)

#### Minimum swift support

Swift 5 drops support for the older Swift 3 package format, so packages that have a Package.swift and Package@swift-4.swift will fail to compile.  This PR renames `Package@swift-4.swift` to `Package.swift`, bringing the minimum supported version to 4.0.

We could instead introduce a `Package@swift-5.swift`, but this increasingly adds complexity for the sake of supporting an obsolete version.

Dropping Swift 3 support should be released under a new SemVer minor, to enable anyone who still requires Swift 3 support to remain on the previous minor version.

## Checklist:

- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -1,34 +1,51 @@
...
 import PackageDescription
 
 #if os(Linux) || os(macOS) || os(iOS) || os(tvOS)
-    
 let package = Package(
     name: "Socket",
-    targets: [Target(name: "Socket")],
-    exclude: ["BlueSocket.xcodeproj", "BlueSocket.xcworkspace", "README.md", "Sources/Info.plist", "Sources/Socket.h"]
+    products: [
+        // Products define the executables and libraries produced by a package, and make them visible to other packages.
```
Comment (ianpartridge) [inline Package.swift]: Can we delete lines like this boilerplate please?

Comment (djones6) [reply Package.swift]: I've pushed an update that removes the stock comments, updates the copyright header to be consistent with the style of other IBM-Swift repos, and also removed the redundant (I believe?) `excludes` from the Socket target.

**Comment by billabt** [comment]
Also, I'd prefer not to merge this PR until a similar PR is presented for _BlueSSLService_.  I'd like to keep them in lockstep.  Thanks.

**Comment by billabt** [comment]
@djones6 I'm under the impression that the _excludes_ will speed up the SPM processing since SPM will know upfront not to process those files.

**Pull request merged**